kubernetes/deployment.yml

```diff
@@ -0,0 +1,205 @@
+apiVersion: apps/v1
```

Comment:
Question: Would we consider to use `kustomize` for managing k8s related config in this project too ?
(the raw config looks a bit difficult to read & learn)

Reply:
kustomize setup and test passed

Reply:
Do we still need `kubernetes/deployment.yml` since kustomize deployment file has been added ?